VSGitDiff/GitDiff.cs

```diff
@@ -36,89 +41,51 @@ internal sealed class GitDiff
         {
-            get;
-            private set;
-        }
+            await package.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken: cancellationToken);
 
-        /// <summary>
-        /// Gets the service provider from the owner package.
-        /// </summary>
-        private IServiceProvider ServiceProvider
-        {
-            get
+            scciProvider = await package.GetServiceAsync(typeof(IVsRegisterScciProvider)) as IVsGetScciProviderInterface;
+
+            var commandService = await package.GetServiceAsync(typeof(IMenuCommandService)) as OleMenuCommandService;
+            if (commandService != null)
             {
-                return this.package;
+                AddMenuCommand(commandService, CmdID_SavedHeadSolution, SavedHeadSolutionCallback, SavedHeadSolutionQuery);
+                AddMenuCommand(commandService, CmdID_WorkingHeadSolution, WorkingHeadSolutionCallback, WorkingHeadSolutionQuery);
+                AddMenuCommand(commandService, CmdID_SavedHeadCodeWin, SavedHeadCodeWinCallback, SavedHeadCodeWinQuery);
+                AddMenuCommand(commandService, CmdID_WorkingHeadCodeWin, WorkingHeadCodeWinCallback, WorkingHeadCodeWinQuery);
+                AddMenuCommand(commandService, CmdID_SavedHeadSourceChanges, SavedHeadSolutionCallback, null);
             }
-        }
 
-        /// <summary>
-        /// Initializes the singleton instance of the command.
-        /// </summary>
-        /// <param name="package">Owner package, not null.</param>
-        public static void Initialize(Package package, ref IVsGetScciProviderInterface _scciProvider)
-        {
-            scciProvider = _scciProvider;
-            Instance = new GitDiff(package);
+            teamExplorer = await package.GetServiceAsync(typeof(ITeamExplorer)) as ITeamExplorer;
+
+            dte = Package.GetGlobalService(typeof(DTE)) as EnvDTE80.DTE2;
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="GitDiff"/> class.
-        /// Adds our command handlers for menu (commands must exist in the command table file)
+        /// Adds a menu command for a given command ID, invocation handler, and status request handler.
         /// </summary>
-        /// <param name="package">Owner package, not null.</param>
-        private GitDiff(Package package)
+        /// <param name="commandService">The service used to add handlers for menu commands.</param>
+        /// <param name="commandID">The numeric command ID.</param>
+        /// <param name="invokeHandler">The event handler called to execute the command.</param>
+        /// <param name="beforeQueryStatusHandler">The event handler called when a client requests the command status.</param>
+        private static void AddMenuCommand(OleMenuCommandService commandService, int commandID, EventHandler invokeHandler, EventHandler beforeQueryStatusHandler)
```

Comment:
Bundle up these calls into one function for clarity.

Note: passing null for the `changeHandler` (and sometimes, `beforeQueryStatusHandler`) is allowed in `OleMenuCommand` and how the different constructors are doing this internal to the class.